git.go

```diff
@@ -230,12 +230,18 @@ func autoMerge(gitCmd git.Git, mergeBranch, branchDest, buildURL, apiToken strin
 	return nil
 }
 
-func manualMerge(gitCmd git.Git, repoURL, prRepoURL, branch, commit, branchDest string) error {
+func manualMerge(gitCmd git.Git, repoURL, prRepoURL, branch, commit, branchDest string, autoMerge bool) error {
 	if err := runWithRetry(func() *command.Model { return gitCmd.Fetch("origin", "refs/heads/"+branchDest) }); err != nil {
 		return fmt.Errorf("fetch failed, error: %v", err)
 	}
-	if err := pull(gitCmd, branchDest); err != nil {
-		return fmt.Errorf("pull failed (%s), error: %v", branchDest, err)
+	if autoMerge {
+		if err := pull(gitCmd, branchDest); err != nil {
+			return fmt.Errorf("pull failed (%s), error: %v", branchDest, err)
+		}
+	} else {
+		if err := run(gitCmd.Checkout(branchDest)); err != nil {
```

Comment:
branchDest = master; is this the correct branch?

Reply:
Maybe $Branch would be the correct branch?

Reply:
branchDest is the existing branch, no changes here just that it does not call 

`run(gitCmd.Merge("origin/" + branchDest))`

Reply:
branchDest is the existing branch, no changes here just that it does not call 

`run(gitCmd.Merge("origin/" + branchDest))`

Reply:
I think that the `autoMerge == true` case is correct but if autoMerge is disabled (else clause) the step will check out master (branchDest) instead of the feature banch (branch).

The PR is opened for merging `branch` -> `branchDest`, e.g.: `feature1` -> `master`.
`autoMerge == true` should check out `feature1 merge into master` and `autoMerge == false` should check out `feature1`. Is that correct @DamienBitrise  @godrei ?

Reply:
Pull() does the following

`if err := run(gitCmd.Checkout(branchDest)); err != nil {
		return fmt.Errorf("checkout failed (%s), error: %v", branchDest, err)
	}
	return run(gitCmd.Merge("origin/" + branchDest))`

So this is all that is different between the if else

`run(gitCmd.Merge("origin/" + branchDest))`

Reply:
So as @DamienBitrise pointed out, branchDest  is the correct one, see: https://bitrise.atlassian.net/wiki/spaces/EnterpriseSales/pages/700514342/Git+Clone+-+Performance+Analysis

At this point, we either check out master or check out master + merge master but what is the value of merging master into master? @godrei @trapacska